- Fix test programs so they don't error out during configure
- Remove hardcoded optimizations.

Index: configure
--- configure.orig
+++ configure
@@ -280,7 +280,7 @@ if [ "${CROSSBUILD}" = 1 ]; then
 if [ "$?" = 0 ]; then CXX="${HOST}-${CXX}"; fi
 fi
 echo "#include <iostream>" > test.cxx
-echo "main(){ std::cout << 1; }" >> test.cxx
+echo "int main(){ std::cout << 1; }" >> test.cxx
 (exec ${CXX} ${CXXFLAGS} ${LDFLAGS} test.cxx -lstdc++ >/dev/null 2>&1)
 if [ $? = 0 ]; then echo ${CXX}; else
 HAVE_LANG_CXX=0
@@ -290,7 +290,7 @@ echo "ERROR: ${CXX} cannot create executables" >&2 ;
 do_remove
 exit 1
  fi
-CXXFLAGS="${CXXFLAGS} -Wall -O2"
+CXXFLAGS="${CXXFLAGS} -Wall"
 CPPFLAGS="${CPPFLAGS} "
 SRCS="${SRCS} tudu.cc data.cc editor.cc interface.cc window.cc text.cc parser.cc config.cc date.cc screen.cc scheduler.cc cmd.cc"
 check_include() {
@@ -300,7 +300,7 @@ _REQUIRED=$3
 printf "checking for $2... "
 echo > test.c
 echo "#include <${INC}>" >>test.c
-echo "main(){}" >>test.c
+echo "int main(){}" >>test.c
 eval \$${COMPILER} ${CFLAGS} ${CPPFLAGS} ${CXXFLAGS} test.c >/dev/null 2>&1
 if [ $? = 0 ]; then
 	eval ${VAR}=1
@@ -337,7 +337,7 @@ _REQUIRED=$3
 _CHKLIB_NAME=$2
 _CHKLIB_LIBS=$(echo "-l${_CHKLIB_NAME}" | sed 's,\+, -l,g')
 printf "checking for lib${_CHKLIB_NAME} ... "
-echo "main(){ }" > test.c
+echo "int main(){ }" > test.c
 eval ${S}${COMPILER} ${CFLAGS} ${CPPFLAGS} ${CXXFLAGS} ${LDFLAGS} ${_CHKLIB_LIBS} test.c >/dev/null 2>&1
 if [ $? = 0 ]; then
 	eval ${VAR}=1
